wsi_deid/web_client/views/ItemView.js

```diff
@@ -102,15 +109,85 @@ wrap(ItemView, 'render', function (render) {
         return isSquare && $(event.target).is('input[type="checkbox"]');
     };
 
-    const showRedactButton = (keyname) => {
-        if (keyname.match(/^internal;aperio_version$/)) {
-            return false;
+    const getFormat = () => {
+        if (this.$el.find('.large_image_metadata_value[keyname^="internal;openslide;openslide.vendor"]').text() === 'aperio') {
+            return formats.aperio;
+        } else if (this.$el.find('.large_image_metadata_value[keyname^="internal;openslide;openslide.vendor"]').text() === 'hamamatsu') {
+            return formats.hamamatsu;
+        } else if (this.$el.find('.large_image_metadata_value[keyname^="internal;xml;PIM_DP_"]').length > 0) {
+            return formats.philips;
+        } else {
+            return formats.none;
         }
-        if (keyname.match(/^internal;openslide;openslide\.(?!comment$)/)) {
+    };
+
+    const isValidRegex = (string) => {
+        try {
+            void new RegExp(string);
+        } catch (e) {
+            console.error(`There was an error parsing "${string}" as a regular expression: ${e}.`);
             return false;
         }
-        if (keyname.match(/^internal;openslide;tiff.(ResolutionUnit|XResolution|YResolution)$/)) {
-            return false;
+        return true;
+    };
+
+    const validateRedactionPatternObject = (redactionPatterns) => {
+        for (const key in redactionPatterns) {
+            if (!isValidRegex(key) || !isValidRegex(redactionPatterns[key])) {
+                delete redactionPatterns[key];
+            }
+        }
+        return redactionPatterns;
+    };
+
+    const getRedactionDisabledPatterns = (settings) => {
+        const format = getFormat();
+        let patterns = settings.no_redact_control_keys; // patterns is an object that looks like {key:value}, where `key` and `value` are both regular expressions
+        switch (format) {
+            case formats.aperio:
+                patterns = Object.assign(patterns, patterns, settings.no_redact_control_keys_format_aperio);
```

Comment:
Won't this change `settings.no_redact_control_keys`?  Shouldn't it be `Object.assign({}, patterns, ...);`?

And, you could avoid the switch statement by doing:
```
patterns = Object.assign({}, patterns, settings[no_redact_control_keys_format_ + format] || {})
```

Reply:
I went ahead and made this change to avoid any unintended side effects. I also liked the idea of removing the switch statement.